rfd/0089-merge-webapps.md

```diff
@@ -0,0 +1,192 @@
+confusion.
+
+## Success Criteria
+
+Both [Teleport](https://github.com/gravitational/teleport) and
+[Teleport.e](https://github.com/gravitational/teleport.e) repos contain their
+respective UI counterparts and the build systems correctly build all projects.
+
+## Developer Experience
+
+### Day to day
+
+The Day to day development experience shouldn't change. The webapps project
+will continue to be self-contained and only require access to the nodejs and
+yarn binaries. The locations of the enterprise assets will have to be changed
+but that should remain transparent to the day to day activities.
+
+To work on the enterprise version of the application the developer will have to
+check out the enterprise version of Teleport which will bring with it the
+appropriate webapp assets.
+
+Merging these repositories may cause some disruption when multiple developers
+are working on the same feature across the back and frontend. Using a
+[git worktree](https://git-scm.com/docs/git-worktree) to create a worktree of
+the webapps folder is an elegant way of approaching this issue
+
+### Backports
+
+To avoid having different proceses and build systems depending on the version
+of teleport, all supported release branches (v9, v10, v11) will have their
+respective versions of the webapps repository.
+
+### Build process
+
+The Teleport build process will need to be expanded to include nodejs and yarn
+in order to be able to build the webapp assets and include into the final
+teleport binaries. We will no longer be committing a compiled version of the
+webassets into the teleport repository, the UI will need to be compiled from
+source on the initial build of teleport. A make target will be provided to
+perform a dockerized build of the webassets for those who don't wish to install
+node and yarn on their workstation.
+
+## Process
+
+Below outlines the process for each repository in the order that the steps need
+to be taken. All repositories will be prepped for the final archival and merging
+but the final switch over steps will not be taken until v11 has been released.
+
+The git histories of each branch will be maintained while merging. [Merging repositories](https://stackoverflow.com/questions/13040958/merge-two-git-repositories-without-breaking-file-history)
+
+### Webapps repository
+
+[repository](https://github.com/gravitational/webapps)
+
+#### Actions
+
+- [ ] Triage issues.
+  - [ ] Close those that are no longer relevant.
+  - [ ] [Transfer remaining issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) to the [Teleport repo](https://github.com/gravitational/teleport)
+  - [ ] Add the [ui label](https://github.com/gravitational/teleport/labels/ui) to
+        those transferred issues.
+- [ ] Triage PRs.
+  - [ ] Close those that are no longer relevant.
+  - [ ] Merge remaining PRs.
+  - [ ] Apply the necessary backports.
```

Comment:
How do we plan to orchestrate this? We cannot tell people to stop working until the switch is performed. But the way these points are stated here doesn't seem to imply that it'll be a seamless process.

Assuming the git history is kept, I think it could be possible to move any WIP branches by adding a remote in your local webapps clone which points to teleport, then push the branch to that remote, clone the branch in your local clone of teleport and rebase the branch on top of already migrated master of teleport.

Reply:
Ultimately the code will be 'moved over' one day so all work can continue on the existing branches until that time. The issue that this is trying to solve is that there are a number of open PR's that are stale so in having a cleaner list of open PR's will make the move simpler.

Reply:
> Ultimately the code will be 'moved over' one day so all work can continue on the existing branches until that time. The issue that this is trying to solve is that there are a number of open PR's that are stale so in having a cleaner list of open PR's will make the move simpler.

Ah, thanks for the explanation, that makes sense. It looks like we were on the same page but I think this point about people being able to continue work on their regular branches wasn't expressed in the RFD so I wasn't sure how you planned around this.